plugins/UM3NetworkPrinting/PrintJobInfoBlock.qml

```diff
@@ -0,0 +1,359 @@
+            bottom: parent.bottom
+            bottomMargin: base.shadowRadius
+        }
+
+        layer.enabled: true
+        layer.effect: DropShadow
+        {
+            radius: base.shadowRadius
+            verticalOffset: 2
+            color: "#3F000000"  // 25% shadow
+        }
+
+        Item
+        {
+            // Content on the left of the infobox
+            anchors
+            {
+                top: parent.top
+                bottom: parent.bottom
+                left: parent.left
+                right: parent.horizontalCenter
+                margins: 2 * UM.Theme.getSize("default_margin").width
+                rightMargin: UM.Theme.getSize("default_margin").width
+            }
+
+            Label
+            {
+                id: printJobName
+                text: printJob.name
+                font: UM.Theme.getFont("default_bold")
+                width: parent.width
+                elide: Text.ElideRight
+            }
+
+            Label
+            {
+                id: ownerName
+                anchors.top: printJobName.bottom
+                text: printJob.owner
+                font: UM.Theme.getFont("default")
+                opacity: 0.6
+                width: parent.width
+                elide: Text.ElideRight
+            }
+
+            Image
+            {
+                source: printJob.previewImageUrl
+                anchors.top: ownerName.bottom
+                anchors.horizontalCenter: parent.horizontalCenter
+                anchors.bottom: totalTimeLabel.bottom
+                width: height
+            }
+
+            Label
+            {
+                id: totalTimeLabel
+                opacity: 0.6
+                anchors.bottom: parent.bottom
+                anchors.right: parent.right
+                font: UM.Theme.getFont("default")
+                text: printJob != null ? getPrettyTime(printJob.timeTotal) : ""
+                elide: Text.ElideRight
+            }
+        }
+
+        Item
+        {
+            // Content on the right side of the infobox.
+            anchors
+            {
+                top: parent.top
+                bottom: parent.bottom
+                left: parent.horizontalCenter
+                right: parent.right
+                margins: 2 * UM.Theme.getSize("default_margin").width
+                leftMargin: UM.Theme.getSize("default_margin").width
+            }
+
+            Label
+            {
+                id: targetPrinterLabel
+                elide: Text.ElideRight
+                font: UM.Theme.getFont("default_bold")
+                text:
+                {
+                    if(printJob.assignedPrinter == null)
+                    {
+                        return catalog.i18nc("@label", "Waiting for: first available")
+                    }
+                    else
+                    {
+                        return catalog.i18nc("@label", "Waiting for: ") + printJob.assignedPrinter.name
```

Comment:
Ideally this would use formatting for the i18n string and not assume so much about how every language should build up this bit of text. Also it really should be combined with the  i18n above. The "Waiting for: " bit is meant to be the same although the translators might not realise it.